### PR TITLE
Consolidate the code that saves the config file

### PR DESCRIPTION
**Pull request by bburtin**
Remove ConfigSubcommand._write_config() and move the code into
CLIConfig.save_config().  Before this change, we created ~/.brkt only
when calling ConfigSubcommand.  Callsites that called
CLIConfig.save_config() would fail when ~/.brkt didn't exist.

### CODE REVIEW

File: brkt_cli/config/__init__.py [hunk 1]
```diff
@@ -325,9 +325,18 @@ def write(self, f):
     def save_config(self):
         """Save the current config to disk.
         """
+        try:
+            os.mkdir(CONFIG_DIR, 0755)
+        except OSError as e:
+            if e.errno != errno.EEXIST:
+                raise
         f = tempfile.NamedTemporaryFile(delete=False, prefix='brkt_cli')
-        yaml.dump(self._config, f)
```
Comment (bonzotronix) [inline brkt_cli/config/__init__.py]: Where do we now dump the self._config YAML to f?

**Comment by bburtin** [comment]
I decided not to write unit tests, because `~/.brkt/config` is hardcoded.  I guess I could generalize the code, but I'm not sure that it's worth the added complexity.

File: brkt_cli/config/__init__.py (hunk 1 above)
Comment (bburtin) [inline brkt_cli/config/__init__.py]: `self.write(f)` below

Comment (bonzotronix) [inline brkt_cli/config/__init__.py]: Ahh, thanks.

**Pull request merged**